.github/actions/slack-notification/slack_payload.py

```diff
@@ -0,0 +1,155 @@
+              }
+          }
+    """
+    return {
+        "type": "section",
+        "text": {
+            "type": "mrkdwn",
+            "text": message,
+        },
+    }
+
+
+def process_message_env():
+    """
+    Process the 'MESSAGE' environment variable for creating a Slack-compatible message.
+
+    If 'URL_ENCODED' is set to 'true', the message is URL-decoded.
+    The message is then formatted as a Slack message containing a single text block.
+
+    Returns:
+    dict: A dictionary representing the Slack message with the processed 'MESSAGE'.
+          The dictionary has the following format:
+          {
+              "text": processed_message,
+              "blocks": [
+                  {
+                      "type": "section",
+                      "text": {
+                          "type": "mrkdwn",
+                          "text": processed_message
+                      }
+                  }
+              ]
+          }
+    """
+    message = os.environ.get("MESSAGE", "No message")
+    message = replace_user_mentions(message, load_users_from_env())
+    if os.environ["URL_ENCODED"] == "true":
+        message = unquote(message)
+    message = "\n".join(line.strip() for line in message.splitlines())
+    return {
+        "text": message,
+        "blocks": [create_message_block(message)],
+    }
+
+
+def set_output(name: str, value: str):
+    """
+    Set an output variable for the GitHub Actions workflow.
+
+    Parameters:
+        name (str): The name of the output variable.
+        value (str): The value to set for the output variable.
+    """
+    with open(os.environ["GITHUB_OUTPUT"], "a", encoding="utf-8") as fh:
+        print(f"{name}={value}", file=fh)
+
+
+def load_users_from_env():
+    """
+    Load a dictionary of users from the 'USERS' environment variable.
+
+    Returns:
+    dict: A dict of users.
+    """
+    users = os.environ.get("USERS")
+    if users is None or users == "":
+        return {}
+    try:
+        return json.loads(users)
+    except json.decoder.JSONDecodeError:
+        print(f"::warning::Invalid value for USERS: {users}")
+        return {}
+
+
+def replace_user_mentions(message, users):
+    """
+    Replace user mentions in the message with the corresponding Slack user IDs.
+
+    Parameters:
+    - message (str): The message to be processed.
+    - users (dict): A dict of users.
+
+    Returns:
+    str: The message with user mentions replaced with Slack user IDs.
+    """
+    for user in users:
+        message = message.replace(f"{user}", f"<@{users[user]}>")
+    return message
+
+
+def main():
+    """
+    Main function to process environment variables and generate a Slack-compatible message.
+    """
+    if os.environ.get("MESSAGE"):
+        json_data = process_message_env()
+    elif os.environ.get("PAYLOAD"):
+        payload = replace_user_mentions(os.environ["PAYLOAD"], load_users_from_env())
+        json_data = json.loads(payload)
```

Comment:
try-catch json.load for safety?

Reply:
I'm not convinced we need it there. If json.loads fails, an exception will occur, and we shouldn't proceed with payload generation.